### PR TITLE
Improve performance around getting stock for Configurable products.

### PR DESCRIPTION
**Pull request by nikunjkotecha**

### CODE REVIEW

File: modules/sku/src/AcquiaCommerce/SKUPluginBase.php [hunk 1]
```diff
@@ -318,14 +318,23 @@ public function getProcessedStock(SKU $sku, $reset = FALSE) {
...
 
     if (!$reset) {
       // Return from Entity field in push mode.
       if ($stock_mode == 'push') {
-        $stock = $sku->get('stock')->getString();
+        if ($sku instanceof SKU) {
+          $stock = $sku->get('stock')->getString();
+        }
+        else {
+          $stock = \Drupal::database()->select('acm_sku_field_data', 'asfd')
```
Comment (miromichalicka) [inline modules/sku/src/AcquiaCommerce/SKUPluginBase.php]: Please inject \Drupal::database through constructor.

File: modules/sku/src/AcquiaCommerce/SKUPluginBase.php [hunk 2]
```diff
@@ -376,6 +385,10 @@ protected function getStock(SKU $sku, $reset = FALSE) {
 
     // Save the value in SKU if we came here as fallback of push mode.
     if ($stock_mode == 'push') {
+      if (!$sku instanceof SKU) {
+        $sku = SKU::loadFromSku($sku_string);
```
Comment (miromichalicka) [inline modules/sku/src/AcquiaCommerce/SKUPluginBase.php:401]: Can you use entityManager for loading?

Comment (nikunjkotecha) [reply modules/sku/src/AcquiaCommerce/SKUPluginBase.php:401]: loadFromSku uses static cache and takes care of translating to current language so using it everywhere, also we'll have sku string (not entity id) here

File: modules/sku/src/AcquiaCommerce/SKUPluginBase.php (hunk 1 above)
Comment (nikunjkotecha) [reply modules/sku/src/AcquiaCommerce/SKUPluginBase.php]: done (for another place in same file as well, IMO we should create public tasks to fix such issues in existing code)

Comment (miromichalicka) [reply modules/sku/src/AcquiaCommerce/SKUPluginBase.php]: definitely, go for it and create as many issue as you want :)

**Pull request merged**